src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2482,7 +2482,7 @@ module Crystal
 
     def visit(node : PointerOf)
       var = pointerof_var(node)
-      node.exp.raise "can't take address of #{node.exp}" unless var
+      node.exp.raise "can't take address of #{node.exp}, put #{node.exp} in a variable" unless var
```

Comment:
This might be incorrect, sometime you simply can't take the address of it I think (I don't have an example OTOH though..)

Reply:
Yes, I think this suggestion is incorrect. Moving things to a variable will give a result that you wouldn't expect. For example if you do `pointerof(self)` and get the error and you move it to a var, and `self` as a struct, then you get a totally different result.

Reply:
If at all, I'd phrase this something like "can't take address of #{node.exp}, pointerof() can only be used with variables"

Reply:
How about `can't take address of #{node.exp}, try putting #{node.exp} in a variable`? In the most cases this works.